test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/BaseOtlpHttpExportClientTests.cs

```diff
@@ -0,0 +1,67 @@
+// <copyright file="BaseOtlpHttpExportClientTests.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Net.Http;
+using OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation.ExportClient;
+using Xunit;
+
+namespace OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests
+{
+    public class BaseOtlpHttpExportClientTests
+    {
+        [Theory]
+        [InlineData(null, null, "http://localhost:4318/signal/path")]
+        [InlineData(null, "http://from.otel.exporter.env.var", "http://from.otel.exporter.env.var/signal/path")]
+        [InlineData("https://custom.host", null, "https://custom.host")]
+        [InlineData("http://custom.host:44318/custom/path", null, "http://custom.host:44318/custom/path")]
+        [InlineData("https://custom.host", "http://from.otel.exporter.env.var", "https://custom.host")]
```

Comment:
I don't think this is anything new you've introduced, so I'm not looking to block your PR on this, but my preference is that environment variables take precedence over code, however, I'm not sure we ever came to a clear agreement on this.

This was discussed a bit here https://github.com/open-telemetry/opentelemetry-dotnet/pull/2395#issuecomment-924091917.

Also, I've spot checked a few other places and we are preferring code over env vars.

@cijothomas and @CodeBlanch, I just want to confirm that your preference is that config via code should take precedence over environment variables.

Reply:
> my preference is that environment variables take precedence over code

1. This would change the existing behavior (IMO breaking change)
2. This would be not consistent with other SDKs. At least the ones that I am aware of (Go, Python).

Reply:
I agree with @pellared: env vars should only override the defaults, then any other value set in code should be respected. It seems the typical pattern for configuration in general: attempt to read from some source(s) while "creating" the config to perhaps override the defaults and then respect what the developer coded.

Reply:
> The current behavior is consistent with other SDKs. At least the ones that I am aware of (Go, Python).

Good to know, I was wondering about this. I also checked with Java and this seems to be the behavior there too. I'm content with these findings - it's a plus that the SIGs have maintained some consistency here.

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry.Exporter.OpenTelemetryProtocol#environment-variables This may not be very obvious from the doc here, would be nice to have a follow up to this doc to avoid any confusion for end users.